interfaces/devdoc/file_requirements.md

```diff
@@ -0,0 +1,116 @@
+# file
+================
+
+## Overview
+
+The `file` module provides a platform-independent API for asynchronous file operations.

```

Comment:
>asynchronous file operations. [](start = 58, length = 29)

Would be nice to list out which operations we support here (read at offset with length, write at position, extend file size)